scylla/src/statement/prepared_statement.rs

```diff
@@ -121,6 +135,13 @@ impl PreparedStatement {
                 buf.extend_from_slice(v);
                 buf.put_u8(0);
             }
+            Ok(())
+        };
+        const ON_STACK_LIMIT: usize = 8;
+        if self.metadata.pk_indexes.len() > ON_STACK_LIMIT {
```

Comment:
This one sounds a little like inventing a small vector from scratch. Maybe we should simply consider pulling in https://crates.io/crates/smallvec? Not only it would make the code above considerably simpler, making the closure redundant, but it would most likely be applicableas an optimization in multiple other places as well. /cc @piodul

Reply:
Pulling in smallvec sounds good.